src/tof/model.py

```diff
@@ -44,6 +42,24 @@ def __init__(
             self.detectors = [self.detectors]
         self.pulse = pulse
 
+    def add(self, component):
```

Comment:
```suggestion
    def insert(self, component):
```
?

Reply:
I think `insert` suggests it is also expecting a positional index where to insert the component. But since we don't care (sorting according to distance from source is performed on the fly), I actually prefer `add`, I think it is more generic.